provider/k8s.go

```diff
@@ -1048,6 +1049,15 @@ func (k8s *K8sOfflineStore) RegisterPrimaryFromSourceTable(id ResourceID, source
 }
 
 func blobRegisterPrimary(id ResourceID, sourceName string, logger *zap.SugaredLogger, store FileStore) (PrimaryTable, error) {
+	sourceUrlExists, err := doesSourceUrlExist(store, sourceName)
```

Comment:
This is amazing, can this be done upstream somewhere so it catches everything that uses a file store? Maybe in the coordinator?

Reply:
It can I think. Let me do think about how iI would do that in a separate PR